### PR TITLE
#275: Update copyright year and GHA to check year

### LINKED ISSUE #275: Update license copyright year (Workflow)
# Problem
The year in MIT License in files is outdated

# Solution
Update year in files and create shell script to run in Github Workflow to check year date in files

### PR DESCRIPTION
**Pull request by slry**
In this PR I updated the copyright year of MIT license and added Workflow that runs script to check MIT license year on push and pull request
@yegor256, please take a look

Closes #275

---

## PR-Codex overview
This PR adds a GitHub Actions workflow to check for outdated copyright years in files based on the specified pattern.

### Detailed summary
- Added a GitHub Actions workflow for copyright check
- Checks for outdated copyright years in files
- Ignores specified files and directories
- Extracts and compares years to the current year
- Outputs files with outdated copyright years

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by yegor256** [comment]
@slry looks good, but do we really need a separate `.sh` file for this? Can't we put everything into the `run:` block in the YAML file?

**Comment by slry** [comment]
@yegor256 done.

**Comment by slry** [comment]
@yegor256 please, take a look at changes, we may merge it.

**Comment by yegor256** [comment]
@maxonfjvipon please, help us review this one

File: test/test_mvnw.js
Comment (maxonfjvipon) [inline test/test_mvnw.js:1]: @slry why do we need these changes?

Comment (slry) [reply test/test_mvnw.js:1]: I guess, my IDE accidentally formatted this file, will revert this one

**Comment by maxonfjvipon** [comment]
@slry before actual review we need all GHA checks are green

**Comment by slry** [comment]
@maxonfjvipon I pulled updates from master and it passed markdownlint GHA

But some of the build actions are failing because GHA cannot find Node of version 12 for arm64
```bash
Error: Unable to find Node version '12' for platform darwin and architecture arm64.
```

I do not think this is the problem with this PR

**Comment by maxonfjvipon** [comment]
@slry you're right, but we won't be able merge your PR without these checks. You either should fix it, or wait until it's resolved by someone else in other PR

**Comment by slry** [comment]
@maxonfjvipon seems like this issue relates to this [post](https://github.blog/changelog/2022-08-09-github-actions-self-hosted-runners-now-support-apple-m1-hardware/)

Right now node12 is not supported by MacOS ARM64, using node16 could solve this issue

What do you think?

**Comment by maxonfjvipon** [comment]
@slry let's try

**Comment by slry** [comment]
@maxonfjvipon I fixed macos problem, but tests still do not pass

It seems like some of the npm packages do not support node12 and requires >=node16
```bash
/home/runner/work/eoc/eoc/node_modules/commander/lib/command.js:380
eOrNameAndArgs = enableOrNameAndArgs ?? 'help [command]';
                                               ^

SyntaxError: Unexpected token '?'
    at wrapSafe (internal/modules/cjs/loader.js:915:16)
    at Module._compile (internal/modules/cjs/loader.js:963:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/runner/work/eoc/eoc/node_modules/commander/index.js:2:21)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
Error: Process completed with exit code 1.
```

**Comment by maxonfjvipon** [comment]
@slry you're right. [`commander`](https://www.npmjs.com/package/commander#options) package uses [Nullish Coalescing Operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) which is supported on Node >= 14.0. Let's try to replace Node 12 with Node 14 in our GHA CI matrix.

**Comment by slry** [comment]
@maxonfjvipon now tests failing on Windows machine, I assume it cannot parse linux commands (maybe removing \ will fix it...)

```bash
      - run: |
          cd itest
          p=0.36.0
          t=0.36.0
          node ../src/eoc.js "--parser=$p" "--home-tag=$t" --batch \
            dataize program
          node ../src/eoc.js "--parser=$p" "--home-tag=$t" --alone \
            --batch dataize program
          node ../src/eoc.js clean
          node ../src/eoc.js "--parser=$p" "--home-tag=$t" --batch \
            test
```

GHA output:
```bash
ParserError: D:\a\_temp\ff659728-8805-4818-afa3-684a0463d1c4.ps1:8
Line |
   8 |    --batch dataize program
     |      ~
     | Missing expression after unary operator '--'.
Error: Process completed with exit code 1.
```

File: .github/workflows/copyright_check.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      - master
+
+jobs:
+  copyright:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run check
+        run: |
+          bash - << 'SCRIPT'
```
Comment (yegor256) [inline .github/workflows/copyright_check.yml:19]: @slry I can't understand, why do you need to run Bash script through this `SCRIPT` file? can't you just directly use the text?

Comment (slry) [reply .github/workflows/copyright_check.yml:19]: @yegor256 this script just do not work in yaml for some reason, so this is a workaround I found

**Comment by slry** [comment]
@maxonfjvipon @yegor256 can we merge this?

File: .github/workflows/copyright_check.yml (hunk 1 above)
Comment (yegor256) [reply .github/workflows/copyright_check.yml:19]: @slry it's a weird workaround. I suggest you try to make it work as all other scripts.

**Comment by yegor256** [comment]
@slry there are merge conflicts, can't merge

**Comment by slry** [comment]
@yegor256 resolved conflicts

**Comment by slry** [comment]
@yegor256 made some changes to script:
1. Removed my workaround `bash - << SCRIPT`
2. Fixed some bugs

**Comment by slry** [comment]
@yegor256 can we merge this?

**Comment by slry** [comment]
@yegor256 can we merge this?

**Comment by slry** [comment]
@yegor256 please take a look at changes

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**